src/api.yaml

```diff
@@ -0,0 +1,127 @@
+      beyond the bounds of the sampled texture.
+    ref:
+      name: GPUAddressMode
+      anchor: '#enumdef-gpuaddressmode'
+    values:
+      repeat: Texture coordinates wrap to the other side of the texture
+      mirror repeat: >
+          Texture coordinates wrap to the other side of the texture, but the
+          texture is flipped when the integer part of the coordinate is odd.
+      clamp to edge: Texture coordinates are clamped between 0.0 and 1.0, inclusive.
+
+  alpha mode:
+    description: >
+      Selects how the contents of the surface will be interpreted when read,
+      when rendered to the screen or used as an image source.
+    ref:
+      name: GPUCanvasAlphaMode
+      anchor: '#GPUCanvasAlphaMode'
+    values:
+      premultiplied: >
+          Read RBGA as premultipled: color values are premultipled by their
+          alpha value.
+      unpremultiplied: >
+          Read RGBA as not premulitipled: Color values are not premultipled by
+          their alpha value
+      opaque: Read RGB as opaque and ignore alpha value
+
+  backend type:
+    description: >
+      Specifies the type of backend to use with Dawn.
+    ref:
+    values:
+      'null': A null backend
+      WebGPU: The WebGPU backend
+      D3D11: The D3D11 backend
+      D3D12: The D3D12 backend
+      metal: The Metal backend
+      vulkan: The Vulkan backend
+      openGL: The OpenGL backend
+      openGLES: The OpenGL ES backend
+
+bitmask:
+  texture usage:
+    description: Determine how a GPUTexture maybe be used after creation.
+    ref:
+      name: GPUTextureUsage
+      anchor: '#namespacedef-gputextureusage'
+    values:
+      none: No usage specified.
+      copy src: The texture can be used as the source of a copy operation.
+      copy dst: The texture can be used as the destination of a copy operation.
+      texture binding: >
+        The texture can be bound for use as a sampled texture in a shader.
+      storage binding: >
+        The texture can be bound for use as a storage texture in a shader.
+      render attachment: >
+        The texture can be used as a colour or depth/stencil attachment in a
+        render pass.
+      present: The texture can be used to "present" the texture on screen.
+
+function:
+  create instance:
+    description: Creates an instance of the Dawn WebGPU system
+    returns:
+      instance: The created %% instance %%.
+    args:
+      descriptor: The %% instance descriptor %% pointer. Maybe `%% null %%`.
+
+  get proc address:
+    description: >
+      Returns the function pointer for the requested WebGPU API method *procName*.
+    returns:
+      proc: The %% proc %% function pointer.
+    args:
+      device: The %% device %% to retrieve the proc for
+      proc name: The name of the pointer to retrieve
+
+structure:
+  adapter properties:
+    description: Property information for a specific adapter
+    values:
+      vendor ID: The ID for the adapter vendor
```

Comment:
I feel like the name are pretty self explanatory, so documentation should go a bit beyond, for example explaining it is a PCIe ID if available but can be other values on system not using PCIe ID for example.

Architecture is a new concept in graphics APIs as well so we could explain what it is, etc.

Reply:
I'll grab a bit more context from the WebGPU spec and add it in. (https://www.w3.org/TR/webgpu/#gpuadapterinfo)

Reply:
done